File: app/src/main/java/com/katkov/lolachievements/application/navigation/Screens.java
package com.katkov.lolachievements.application.navigation;

import com.katkov.lolachievements.application.ui.checkentryinfo.CheckEntryInfoFragment;
import com.katkov.lolachievements.application.ui.firstentry.FirstEntryFragment;
import com.katkov.lolachievements.application.ui.summonerinfo.SummonerInfoFragment;

import androidx.fragment.app.Fragment;
import ru.terrakok.cicerone.android.support.SupportAppScreen;

public class Screens {

    public static class FirstEntryScreen extends SupportAppScreen {
        @Override
        public Fragment getFragment() {
            return FirstEntryFragment.newInstance();
        }
    }

    public static class CheckFirstEntryInfoScreen extends SupportAppScreen {
        @Override
        public Fragment getFragment() {
            return CheckEntryInfoFragment.newInstance();
        }
    }

    public static class SummonerInfoScreen extends SupportAppScreen {

        @Override
        public Fragment getFragment() {
            return SummonerInfoFragment.newInstance();
        }
    }
}

File: app/src/main/java/com/katkov/lolachievements/application/navigation/Screens.kt
package com.katkov.lolachievements.application.navigation

import com.katkov.lolachievements.application.ui.achievements.AchievementsFragment
import com.katkov.lolachievements.application.ui.bottomnavigation.BottomNavigationFragment
import com.katkov.lolachievements.application.ui.checkentryinfo.CheckEntryInfoFragment
import com.katkov.lolachievements.application.ui.login.LoginFragment
import com.katkov.lolachievements.application.ui.summonerinfo.SummonerInfoFragment
import ru.terrakok.cicerone.android.support.SupportAppScreen

class Screens {

    class LoginScreen : SupportAppScreen() {
        override fun getFragment() = LoginFragment.newInstance()
    }

    class BottomNavigationFragmentScreen : SupportAppScreen() {
        override fun getFragment() = BottomNavigationFragment.newInstance()
    }

    class AchievementsScreen : SupportAppScreen() {
        override fun getFragment() = AchievementsFragment.newInstance()
    }

    class CheckFirstEntryInfoScreen : SupportAppScreen() {
        override fun getFragment() = CheckEntryInfoFragment.newInstance()
    }

    class SummonerInfoScreen : SupportAppScreen() {
        override fun getFragment() = SummonerInfoFragment.newInstance()
    }
}
